runtime/major_gc.c

```diff
@@ -1703,8 +1769,16 @@ static void major_collection_slice(intnat howmuch,
     if (log_events) CAML_EV_END(EV_MAJOR_SWEEP);
   }
 
+  if (domain_state->sweeping_done) {
+    /* We do not immediately trigger a minor GC, but instead wait for
```

Comment:
As soon as one of the domains finishes sweeping, it requests a mark phase. However, we can potentially delay the marking phase further. Is there any reason to consider doing something different here?

Reply:
There is definitely more work to do here to perform well in the case where sweeping finishes at different times in different domains. One route I'd like to explore is balancing sweep work between domains.

For now, though, this behaviour is reasonably close to what trunk does (begins marking mode as soon as any domain would otherwise begin marking), so should not be worse than current behaviour.

Reply:
A good question. Any domain which has finished sweeping can't do any major GC work until marking starts, whereas any domain still sweeping can continue sweeping after marking has started. But conceivably the gains from additionally delaying the start of marking could outweigh the costs of a domain twiddling its thumbs during GC pauses.

Reply:
I've added a TODO comment here.